synapse/handlers/federation_event.py

```diff
@@ -380,7 +377,7 @@ async def on_send_membership_event(
         # need to.
         await self._event_creation_handler.cache_joined_hosts_for_event(event, context)
 
-        await self._check_for_soft_fail(event, None, origin=origin)
+        await self._check_for_soft_fail(event, context=context, origin=origin)
```

Comment:
We have all the prev events here, otherwise we would have raised a `RuntimeError` when computing the event context. Thus the soft fail check will behave the same as before.